transformer_engine/common/common.cu

```diff
@@ -97,7 +97,15 @@ void create_2D_tensor_map(CUtensorMap &tensorMap, const SimpleTensor &tensor,
 
   const CUtensorMapDataType tensorDataType = get_CUtensorMapDataType(tensor.dtype);
   void *dataPtr = reinterpret_cast<void *>(tensor.dptr);
-  NVTE_CHECK(isPointerAligned(dataPtr, 16), "Tensor data must be 16B aligned");
+
+  constexpr int TMA_gmem_alignment = 16;  // Alignment of the global memory address
+  NVTE_CHECK(isPointerAligned(dataPtr, TMA_gmem_alignment),
+             "Tensor data pointer must be 16B aligned");
+
+  const int TMA_needed_size = TMA_gmem_alignment / type_size;
+  NVTE_CHECK(globalX % TMA_needed_size == 0, "Shape not supported. Expected multiple of " +
+                                                 std::to_string(TMA_needed_size) + ", got " +
+                                                 std::to_string(globalX));
```

Comment:
`NVTE_CHECK` already converts to string internally. It can even print out `std::vector<T>`.
```suggestion
  const int TMA_needed_size = TMA_gmem_alignment / type_size;
  NVTE_CHECK(globalX % TMA_needed_size == 0, "Shape not supported. Expected multiple of " +
                                                 TMA_needed_size + ", got " +
                                                 globalX);
```

Reply:
Ah, it should be something like:
```suggestion
  const int TMA_needed_size = TMA_gmem_alignment / type_size;
  NVTE_CHECK(globalX % TMA_needed_size == 0, "Shape not supported. Expected multiple of ",
                                             TMA_needed_size, ", got ", globalX);
```
This is not high priority, but taking advantage of these convenience features makes it easier to write detailed error messages.